### ISSUE TITLE
Token Request App

### ISSUE THREAD

**Issue by burrrata** [1Hive/grants#2]
The Token Request app allows Aragon organizations to allow users to propose minting tokens in exchange for depositing tokens into the organizations vault. It is not currently under development, but there is a repo for it here:
- https://github.com/1Hive/token-request-app

@lkngtn and @burrrata shared lots of thoughts on this [here](https://github.com/1Hive/ideas/issues/3) and [here](https://github.com/1Hive/website/issues/17). 
TL;DR: @burrrata says yes and @lkngtn says maybe.

@0xGabi said: "This app is really important to allow donations. I think is a must in the proposal"

<br>

It sounds like we generally think it's useful, but it's not as important as the [Redemptions App](#1 ). Is that about accurate?

**Comment by lkngtn** [comment]
To clarify my position! 

I'm very much for the creation of this App. I'm less convinced that it fits into the long term model for 1hive (as I think it makes more sense for us to adopt apiary). I think orgs may choose to start with an app like token-request and then transition to apiary (sort of like going public), but once they are using apiary they probably would uninstall (not currently possible) or atleast disable the token request app.

So my position is I would love for us to build it either as part of a grant #6 proposal, or just because we think it would be useful, and we feel like doing it. But I'm not sure 1hive specifically will end up using it much if at all before being able to transition over to Apiary. 

edit: I also pushed back a bit when the original repo was created because I felt like these discussions we are having now about strategy and direction were important to have as a group before going full steam ahead, not because I didn't support the idea of building the token-request app. :)

**Comment by yeqbfgxjiq** [comment]
> I'm less convinced that it fits into the long term model for 1hive (as I think it makes more sense for us to adopt apiary).

I agree with this 100%. Apiary is the future. Seems like TRA is useful tho and can be implemented sooner than later though, so that's a win too. 

Also agree that it makes sense to include TRA in #6, but move 1Hive to Apiary in the future.

Also also, I appreciate your push back as it helped refine ideas and direction. I'm much happier with the cohesive package of #6 than just barreling through 1 app at a time :)

**Comment by yeqbfgxjiq** [comment]
### Open Questions For All BEEs 🐝
- Is this something we would want to work on?
- Do we agree that these apps are simple and straightforward to develop?
- What is the expected scope and a reasonable timeline given our resources and other commitments?

**Comment by 0xGabi** [comment]
>I think orgs may choose to start with an app like token-request and then transition to apiary (sort of like going public)

Really interesting!

**Comment by willjgriff** [comment]
- Is this something we would want to work on?

I would be happy to work on this although it's necessity sounds like it depends on how far along apiary is. Perhaps it's not necessary if we can use that.

- Do we agree that these apps are simple and straightforward to develop?

This app will have a basic UI with only a few functions similar to the Redemptions app. I can't think of many unknowns with this app, should be straight forward to build.

- What is the expected scope and a reasonable timeline given our resources and other commitments?

I would estimate roughly 2 weeks of work for one developer for a functional tested version with a UI.
